### PR TITLE
Added harmonic mean and changes to stream

### PR DESCRIPTION
**Pull request by SSapkal**
Modified stream benchmark to include:
1) Harmonic mean calculation
2) Included iterations and loop

Signed-off-by: Swapnil Sapkal <swapnil.sapkal@amd.com>

### CODE REVIEW

**Comment by SSapkal** [comment]
Previous ```stats.csv``` Output:
```
stream.element_size_byte,8
stream.array_size,10000000
stream.copy_bandwidth_MBps,31280.4
stream.scale_bandwidth_MBps,22237.7
stream.add_bandwidth_MBps,32653.2
stream.triad_bandwidth_MBps,31418.0
```

Updated ```stats.csv``` Output:

```
stream.element_size_byte,8
stream.array_size,10000000
stream.copy_bandwidth_MBps,20900.35
stream.copy_bandwidth_MBps_harmonicMean,20900.13
stream.scale_bandwidth_MBps,20324.22
stream.scale_bandwidth_MBps_harmonicMean,20323.65
stream.add_bandwidth_MBps,21832.45
stream.add_bandwidth_MBps_harmonicMean,21832.31
stream.triad_bandwidth_MBps,21690.71
stream.triad_bandwidth_MBps_harmonicMean,21690.56
```

```lkp ncompare``` Output:
```
lkp ncompare -s testcase=stream/commit=4.18.0-348.el8.x86_64 -s testcase=stream/commit=5.19.0+ -a -k
=========================================================================================
array_size/compiler/iterations/kconfig/loop/nr_threads/rootfs/tbox_group/testcase:
  10000000/gcc/10x/defconfig/10/100%/rhel/swapnil_rhel85/stream

commit:
  4.18.0-348.el8.x86_64
  5.19.0+

4.18.0-348.el8.x                     5.19.0+
---------------- ---------------------------
         %stddev     %change         %stddev
             \          |                \
     25625 ±  3%      -8.7%      23408 ± 11%  stream.add_bandwidth_MBps
     25616 ±  3%      -9.0%      23302 ± 11%  stream.add_bandwidth_MBps_harmonicMean
  10000000            +0.0%   10000000        stream.array_size
     21572            -3.0%      20921 ±  4%  stream.copy_bandwidth_MBps
     21571            -3.1%      20898 ±  4%  stream.copy_bandwidth_MBps_harmonicMean
      8.00            +0.0%       8.00        stream.element_size_byte
     21335            -4.5%      20386 ±  4%  stream.scale_bandwidth_MBps
     21333            -4.6%      20347 ±  4%  stream.scale_bandwidth_MBps_harmonicMean
     25413 ±  3%      -8.3%      23308 ± 11%  stream.triad_bandwidth_MBps
     25404 ±  3%      -8.6%      23217 ± 11%  stream.triad_bandwidth_MBps_harmonicMean
```

File: stats/stream [hunk 1]
```diff
@@ -3,8 +3,10 @@
 LKP_SRC = ENV['LKP_SRC'] || File.dirname(File.dirname(File.realpath($PROGRAM_NAME)))
 
 require "#{LKP_SRC}/lib/log"
+require "#{LKP_SRC}/lib/statistics"
 
 stats = {}
+stats.default_proc = proc { |hash, key| hash[key] = [] }
```
Comment (rli9) [inline stats/stream:9]: above 2 lines can be combined like stats = Hash.new { |h, k| h[k] = [] }

File: jobs/stream.yaml [hunk 2]
```diff
@@ -6,11 +6,21 @@ nr_threads:
...
     - 10000000 # 0.2GB
     - 50000000 # 1GB
+    - 128000000 # 3GB
+    - 192000000 # 4.5GB
   omp:
     - true
     - false
+  loop:
+    - 10
+    - 100
```
Comment (rli9) [inline jobs/stream.yaml:26]: prefer to only have 100 here, which is default value in original script. If you need further customize the testing like loop 10 or nr_thread, you can consider to maintain your own jobs.

File: jobs/stream.yaml [hunk 3]
```diff
@@ -6,11 +6,21 @@ nr_threads:
...
+
+iterations:
+  - 10x
 
 stream:
   array_size:
     - 10000000 # 0.2GB
     - 50000000 # 1GB
+    - 128000000 # 3GB
+    - 192000000 # 4.5GB
```
Comment (rli9) [inline jobs/stream.yaml:20]: for the newly add array_size, is it possible to maintain this in your own version of job yamls, so we still keep the default in lkp-tests.

**Comment by rli9** [comment]
thanks, besides the code change, would you also add spec for stats to show the stats script work as expected.

File: stats/stream (hunk 1 above)
Comment (SSapkal) [reply stats/stream]: The change is done according to the review comment.

File: jobs/stream.yaml (hunk 2 above)
Comment (SSapkal) [reply jobs/stream.yaml:21]: The changes are done according to the review comments.

**Comment by SSapkal** [comment]
Attaching updated ```stats.csv``` file. 
[stats.csv](https://github.com/intel/lkp-tests/files/9544999/stats.csv)
Please let me know if you want any additional data.

File: jobs/stream.yaml (hunk 3 above)
Comment (SSapkal) [reply jobs/stream.yaml]: The change is updated according to the review comment.

**Comment by rli9** [comment]
> Attaching updated `stats.csv` file. [stats.csv](https://github.com/intel/lkp-tests/files/9544999/stats.csv) Please let me know if you want any additional data.

Thanks, i mean you need add test code under spec dir, you can refer to spec/stats/fio to add new input like 01 or harmonic

**Comment by SSapkal** [comment]
> > Attaching updated `stats.csv` file. [stats.csv](https://github.com/intel/lkp-tests/files/9544999/stats.csv) Please let me know if you want any additional data.
> 
> Thanks, i mean you need add test code under spec dir, you can refer to spec/stats/fio to add new input like 01 or harmonic

Thanks for the clarification, Added the test for stats/stream under spec dir.

**Comment by rli9** [comment]
thanks for the patch

**Pull request merged**